crates/networking/rpc/engine/payload.rs

```diff
@@ -689,45 +751,35 @@ fn validate_fork(block: &Block, fork: Fork, context: &RpcApiContext) -> Result<(
     Ok(())
 }
 
-fn build_execution_payload_response(
+fn build_payload_if_necessary(
     payload_id: u64,
-    payload: (Block, U256, BlobsBundle, bool),
-    should_override_builder: Option<bool>,
+    mut payload: PayloadBundle,
     context: RpcApiContext,
-) -> Result<ExecutionPayloadResponse, RpcErr> {
-    let (mut payload_block, block_value, blobs_bundle, completed) = payload;
-    if completed {
-        Ok(ExecutionPayloadResponse {
-            execution_payload: ExecutionPayload::from_block(payload_block),
-            block_value,
-            blobs_bundle: Some(blobs_bundle),
-            should_override_builder,
-        })
+) -> Result<PayloadBundle, RpcErr> {
+    if payload.completed {
+        Ok(payload)
     } else {
-        let (blobs_bundle, block_value) = {
-            let syncer = context
-                .syncer
-                .try_lock()
-                .map_err(|_| RpcErr::Internal("Error locking syncer".to_string()))?;
-            syncer
-                .blockchain
-                .build_payload(&mut payload_block)
-                .map_err(|err| RpcErr::Internal(err.to_string()))?
+        let syncer = context
+            .syncer
+            .try_lock()
+            .map_err(|_| RpcErr::Internal("Error locking syncer".to_string()))?;
+        let (blobs_bundle, requests, block_value) = syncer
+            .blockchain
```

Comment:
don't access `blockchain` from the syncer, see: https://github.com/lambdaclass/ethrex/pull/2123